docs/troubleshooting/02_rpcerror.md

```diff
@@ -0,0 +1,72 @@
+When a call to the chain_api is performed and fails, it will result in an RPCError object being generated to inform why the transaction failed.
+
+The RPCError object will error with the concise message, for instance 'Invalid transaction', but additional details can be found in both the `details` field and the `json` field.  The `json` field holds the entire json response from `nodeos` while the `details` field specifically holds the error object in the `json`.  This is typically in different places of the `json` depending on what endpoint is used in `nodeos` so the `details` field is available to quickly see the error information without needing to search through the `json` object.
+
+As you can see below, the concise error message might not give enough information to really discern the issue.  While the error has `eosio_assert_message assertion failure` as the concise error, in the `details` field the `overdrawn balance` error mentioned.  It's important to note that the format of the `details` object can be different depending on what endpoint is used.
```

Comment:
Maybe 

As you can see below, the concise error message might not give enough information to really discern the issue. While the error has `eosio_assert_message assertion failure` as the concise error, in the `details` field the `overdrawn balance` error mentioned. It's important to note that the format of the `details` object can be different depending on what endpoint is used.

to =>

In the `details` and `json` examples below, you can see that the error message may not contain enough information to discern what caused the action to fail. The error message contains `eosio_assert_message assertion failure`.  Looking further at the `details` you can see  an `overdrawn balance` message. 

It's important to note that the format of the `details` object can be different depending on what endpoint is used. => removed this bit as it is mentioned above. If you wanted to emphasise (or I have misunderstood) pwerhaps   
=> Please note that the format of the `details` object will differ dependant on which endpoint is used.

=> as an additional thing - I think you choose the endpoint with the readOnlyTrx flag, is it worth mentioning the other endpoint explicitly and/or showing an example from calls to both endpoints -assuming there are two endpoints.

Reply:
There are a large amount of different responses from nodeos, unfortunately it won't be easy to state all of them.